charts/backstage/templates/_helpers.tpl

```diff
@@ -7,7 +7,15 @@ Returns custom hostname
     {{- else if .Values.global.clusterRouterBase -}}
         {{- printf "%s-%s.%s" (include "common.names.fullname" .) .Release.Namespace .Values.global.clusterRouterBase -}}
     {{- else -}}
-        {{ fail "Unable to generate hostname" }}
+        {{/*
+        Attempt to obtain a fallback value for the hostname from the openshift cluster if both global.host and global.clusterRouterBase are ""
+        */}}
+        {{- $cluster := (lookup "config.openshift.io/v1" "Ingress" "" "cluster") -}}
```

Comment:
Okay, I confirmed this call will not work on non-OpenShift clusers:

```
$ helm upgrade --install my-backstage-from-helm charts/backstage \
     --set upstream.backstage.image.tag=next \
     --set route.enabled=false \
     --set global.clusterRouterBase=""

Release "my-backstage-from-helm" does not exist. Installing it now.
lookup_func.go:111: [ERROR] unable to retrieve resource list for: config.openshift.io/v1 , error: the server could not find the requested resource
lookup_func.go:85: [ERROR] unable to get apiresource from unstructured: config.openshift.io/v1, Kind=Ingress , error the server could not find the requested resource
Error: template: backstage/charts/upstream/templates/backstage-deployment.yaml:36:32: executing "backstage/charts/upstream/templates/backstage-deployment.yaml" at <include "common.tplvalues.render" (dict "value" .Values.backstage.appConfig "context" $)>: error calling include: template: backstage/charts/common/templates/_tplvalues.tpl:19:8: executing "common.tplvalues.render" at <tpl $value .context>: error calling tpl: error during tpl function execution for "app:\n  baseUrl: https://{{- include \"janus-idp.hostname\" . }}\nbackend:\n  auth:\n    keys:\n    - secret: ${BACKEND_SECRET}\n  baseUrl: https://{{- include \"janus-idp.hostname\" . }}\n  cors:\n    origin: https://{{- include \"janus-idp.hostname\" . }}\n  database:\n    connection:\n      password: ${POSTGRESQL_ADMIN_PASSWORD}\n      user: postgres": template: backstage/charts/upstream/templates/backstage-deployment.yaml:2:23: executing "backstage/charts/upstream/templates/backstage-deployment.yaml" at <include "janus-idp.hostname" .>: error calling include: template: backstage/templates/_helpers.tpl:13:25: executing "janus-idp.hostname" at <lookup "config.openshift.io/v1" "Ingress" "" "cluster">: error calling 
lookup: unable to get apiresource from unstructured: config.openshift.io/v1, Kind=Ingress: the server could not find the requested resource
```

Reply:
I initially left it as is, since it would fail regardless in non-OpenShift clusters when both `global.host` and `global.clusterRouterBase` fields are missing. But I can change it to make the error more verbose, so users will know what the issue is.